### ISSUE TITLE
All CSD tagged articles getting added to watchlist, despite selecting "don't add to watchlist"

### ISSUE THREAD

**Issue by NovemLinguae** [wikimedia-gadgets/twinkle#1427]
Pretty much the title. Here's a screenshot of my settings.

![image](https://user-images.githubusercontent.com/79697282/125159382-e9f62c80-e12b-11eb-8fa8-d64be8fc1538.png)

**Comment by NovemLinguae** [comment]
Confirmed that unchecking all the boxes still has the bug.

Example: I tagged these pages G5 on 11/1/21, and they were permanently added to my watchlist: [[Hamid Soltani]], [[Max Muay Thai]] 

![image](https://user-images.githubusercontent.com/79697282/139641621-a9dddf81-0b2e-4a86-b656-84ac39e84796.png)

**Comment by siddharthvp** [comment]
@NovemLinguae would you like to work on this bug?

**Comment by NovemLinguae** [comment]
@siddharthvp Sure, you can assign to me. I can take a look in about a week after my day job calms down. Do we have a test wiki?

**Comment by siddharthvp** [comment]
Yes, there's https://test.wikipedia.org. You can clone the repo and run `npm start` to load the gadget from the local source files.
